phidgets/phidget.go

```diff
@@ -129,3 +142,44 @@ func (p *phidget) GetChannelClassName() (string, error) {
 func (p *phidget) SetIsHubPortDevice(b bool) error {
 	return p.phidgetError(C.Phidget_setIsHubPortDevice(p.handle, boolToCInt(b)))
 }
+
+//GetDeviceLabel retrieves the label for the device of this handle
+func (p *phidget) GetDeviceLabel() (string, error) {
+	var cstr *C.char
+	if cerr := C.Phidget_getDeviceLabel(p.handle, &cstr); cerr != C.EPHIDGET_OK {
+		return "", p.phidgetError(cerr)
+	}
+	return C.GoString(cstr), nil
+}
+
+// GetChannelName retrieves the channel name of this handle
+func (p *phidget) GetChannelName() (string, error) {
+	var cstr *C.char
+	if cerr := C.Phidget_getChannelName(p.handle, &cstr); cerr != C.EPHIDGET_OK {
+		return "", p.phidgetError(cerr)
+	}
+	return C.GoString(cstr), nil
```

Comment:
There's no memory leak here right? `cstr` is just a Go pointer on the stack where `C.Phdget_getChannelName` gives us an address? So Go frees the additional address pointer (`cstr`) automatically.

Reply:
No, `C.GoString` does not take ownership of the underlying memory, I think it duplicate it, so it is up to the caller to make sure things are free'd properly. However `C.Phidget_getChannelName` does not returned a malloc'd pointer, it is just a static string, so there is nothing to free.